File: memory-domain/src/main/java/com/memory/dto/relationship/response/RelationshipListResponse.java
package com.memory.dto.relationship.response;

import com.memory.domain.relationship.Relationship;

import java.util.List;
import java.util.stream.Collectors;

public record RelationshipListResponse(
        List<RelationshipResponse> relationships
) {
    public static RelationshipListResponse fromEntities(List<Relationship> relationships) {
        List<RelationshipResponse> relationshipResponses = relationships.stream()
                .map(RelationshipResponse::from)
                .collect(Collectors.toList());
        return new RelationshipListResponse(relationshipResponses);
    }

}

File: memory-api/src/main/kotlin/com/memory/dto/relationship/response/RelationshipListResponse.kt
package com.memory.dto.relationship.response

import com.memory.domain.relationship.Relationship
import java.util.stream.Collectors

data class RelationshipListResponse(
    val relationships: MutableList<RelationshipResponse?>?
) {
    companion object {
        @JvmStatic
        fun fromEntities(relationships: List<Relationship>): RelationshipListResponse {
            return RelationshipListResponse(
                relationships.stream()
                    .map { relationship: Relationship -> RelationshipResponse.from(relationship) }
                    .collect(Collectors.toList())
            )
        }
    }
}
